eos-paygd/main.c

```diff
@@ -35,8 +35,95 @@
 #include <sys/un.h>
 #include <libeos-payg/efi.h>
 
+#define LOGFILE_DIRNAME "/var/log/eos-payg"
+#define LOGFILE_BASENAME "eos-paygd"
+#define LOGFILE_EXT "log"
+
 #define TIMEOUT_POWEROFF_ON_ERROR_MINUTES 20
 
+static gboolean
+open_log_file (GIOChannel **log_io)
+{
+  const gchar *log_file_name = NULL;
+  g_autoptr(GFile) log_file = NULL;
+  g_autoptr(GDateTime) now = NULL;
+  g_autofree gchar *tstamp = NULL;
+  g_autoptr(GError) error = NULL;
+
+  /* Build log file name with a date stamp so we get one file per day,
+   * ex., eos-paygd-20220418.log
+   */
+  now = g_date_time_new_now_local ();
+  tstamp = g_date_time_format (now, "%Y%m%d");
+  log_file_name = g_strconcat (LOGFILE_DIRNAME, "/", LOGFILE_BASENAME, "-",
+                               tstamp, "." LOGFILE_EXT, NULL);
```

Comment:
```suggestion
                               tstamp, ".", LOGFILE_EXT, NULL);
```
Oops! (Still works due to the magical string literal concatenation performed by cpp).